posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -0,0 +1,654 @@
+
+    saved_query = await database_sync_to_async(DataWarehouseSavedQuery.objects.filter(team=team, **filter_params).get)()
+
+    query_columns = saved_query.columns
+    if not query_columns:
+        query_columns = await database_sync_to_async(saved_query.get_columns)()
+
+    table_columns = {}
+    for column_name, column_info in query_columns.items():
+        clickhouse_type = column_info["clickhouse"]
+        nullable = False
+
+        if nullable_match := re.match(NullablePattern, clickhouse_type):
+            clickhouse_type = nullable_match.group(1)
+            nullable = True
+
+        clickhouse_type = re.sub(r"\(.+\)+", "", clickhouse_type)
+
+        table_columns[column_name] = {
+            "data_type": CLICKHOUSE_DLT_MAPPING[clickhouse_type],
+            "nullable": nullable,
+        }
+
+    hogql_query = saved_query.query["query"]
+
+    destination = get_dlt_destination()
+    pipeline = dlt.pipeline(
+        pipeline_name=f"materialize_model_{model_label}",
+        destination=destination,
+        dataset_name=f"team_{team.pk}_model_{model_label}",
+    )
+    _ = await asyncio.to_thread(pipeline.run, hogql_table(hogql_query, team, saved_query.name, table_columns))
+
+    tables = get_delta_tables(pipeline)
+    key, delta_table = tables.popitem()
+    return (key, delta_table)
+
+
+@dlt.source(max_table_nesting=0)
+def hogql_table(query: str, team: Team, table_name: str, table_columns):
+    """A dlt source representing a HogQL table given by a HogQL query."""
+
+    async def get_hogql_rows():
+        response = await asyncio.to_thread(execute_hogql_query, query, team)
+
+        if not response.columns:
+            raise EmptyHogQLResponseColumnsError()
+
+        columns: list[str] = response.columns
+
+        for row in response.results:
+            yield dict(zip(columns, row))
+
+    yield dlt.resource(
+        get_hogql_rows,
+        name="hogql_table",
+        table_name=table_name,
+        table_format="delta",
+        write_disposition="merge",
+        columns=table_columns,
+    )
+
+
+def get_dlt_destination():
+    if TEST:
+        credentials = {
+            "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
+            "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
+            "endpoint_url": settings.OBJECT_STORAGE_ENDPOINT,
+            "region_name": settings.AIRBYTE_BUCKET_REGION,
+            "AWS_ALLOW_HTTP": "true",
+            "AWS_S3_ALLOW_UNSAFE_RENAME": "true",
+        }
+    else:
+        credentials = {
+            "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
+            "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
+            "region_name": settings.AIRBYTE_BUCKET_REGION,
+            "AWS_S3_ALLOW_UNSAFE_RENAME": "true",
+        }
+
+    return dlt.destinations.filesystem(
+        credentials=credentials,
+        bucket_url=settings.BUCKET_URL,  # type: ignore
+        layout="modeling/{table_name}/{load_id}.{file_id}.{ext}",
```

Comment:
Whats the thinking around this layout param? It's not something we use for importing data